src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,419 @@
+                            stInfo.Param = node.Attributes["parameterNames"].Value;
+                        }
+                        ParseSequence(node.ChildNodes, stInfo);
+                    }
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static void ParseSequence(XmlNodeList xn, StackTraceInfo stInfo)
+        {
+            try
+            {
+                foreach (XmlNode node in xn)
+                {
+                    if (node.Name == "sequencePoints")
+                    {
+                        ParseEntry(node.ChildNodes, stInfo);
+                    }
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static int HexToInt(string value)
+        {
+            // strip the leading 0x
+            if (value.StartsWith("0x", StringComparison.OrdinalIgnoreCase))
+            {
+                value = value.Substring(2);
+            }
+            return Int32.Parse(value, NumberStyles.HexNumber);
+        }
+
+        private static void ParseEntry(XmlNodeList xn, StackTraceInfo stInfo)
+        {
+            try
+            {
+                XmlNode bestPointSoFar = null;
+                int ilOffset = HexToInt(stInfo.Offset);
+                foreach (XmlNode node in xn)
+                {
+                    // If the attribute is not 'startLine', but 'hidden', select the best value so far
+                    if (HexToInt(node.Attributes["offset"].Value) > ilOffset)
+                    {
+                        break;
+                    }
+                    if (node.Attributes["startLine"] != null)
+                    {
+                        bestPointSoFar = node;
+                    }
+                }
+                if (bestPointSoFar != null)
+                {
+                    stInfo.StartLine = bestPointSoFar.Attributes["startLine"].Value;
+                    stInfo.EndLine = bestPointSoFar.Attributes["endLine"].Value;
+                    stInfo.Document = bestPointSoFar.Attributes["document"].Value;
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        public static Command SymbolicateCommand() =>
+            new Command(
+                name: "symbolicate", description: "Get the line number from the Method Token and IL Offset at the stacktrace")
+            {
+                // Handler
+                HandlerDescriptor.FromDelegate((SymbolicateDelegate)Symbolicate).GetCommandHandler(),
+                // Arguments and Options
+                InputArgument(),
+                SearchDirectoryOption(),
+                OutputOption()
+            };
+
+        public static Argument InputArgument() =>
```

Comment:
Changing the return type to `Argument<FileInfo[]>` is useful if you want to pass the argument to methods like `ParseResult.ValueForArgument(arg)` and get type inference for the return value.

Reply:
I modified it.